### PR TITLE
Load resource from jars using classloader in TestFileBasedSystemAccessControl

### PR DESCRIPTION
**Pull request by mshang816**
When running test with 'buck test', all test code and related resources will be packaged into a jar file. So any test code that expects resources (test files mostly) from a flat file system will fail, TestFileBasedSystemAccessControl is one of such tests. With this PR, TestFileBasedSystemAccessControl will try to load the resource from a classloader if it's coming from a jar file. Going forward, it's probably a best practice for all our test code to load resource this way to be compatible with different test runners.
The change:
Refactor getResourcePath() int getResourceFile() which will always get the file using getResourceAsStream() and copy its content into a temp file for the tests to use. Then getResourceFile() will work in both cases either running from file systems or jar.

Test plan - (Please fill in how you tested your changes)
Run the test using buck test to make sure all resources can be loaded with no problem.
In the mean time, there should be no existing cases being impacted. So we have to make sure all current CI workflows are green.

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

**Review by ericyuliu — APPROVED** [review]
Thanks for fixing this!

File: presto-main/src/test/java/com/facebook/presto/security/TestFileBasedSystemAccessControl.java [hunk 1]
```diff
@@ -533,7 +537,32 @@ private AccessControlManager newAccessControlManager(TransactionManager transact
 
     private String getResourcePath(String resourceName)
```
Comment (arunthirupathi) [inline presto-main/src/test/java/com/facebook/presto/security/TestFileBasedSystemAccessControl.java:538]: All these are used in copy(source, destination). I believe the code can be simplified if we have a method for copy and if there is an exception, fall into this.

Comment (mshang816) [reply presto-main/src/test/java/com/facebook/presto/security/TestFileBasedSystemAccessControl.java]: @arunthirupathi Refactor the code a bit for a mode concise implementation of getResourceFile() and get rid of getResourcePath(). Please see if you have further comments

**Review by arunthirupathi — APPROVED** [review]
Thanks for making this change, the code looks much more cleaner now.

**Pull request merged**